xbmc/utils/ActorProtocol.h

```diff
@@ -51,7 +51,9 @@ class Message
   bool Reply(int sig, void *data = NULL, int size = 0);
 
 private:
-  Message() {isSync = false; data = NULL; event = NULL; replyMessage = NULL;};
+  Message()
+  : signal(0), isSync(false), isSyncFini(false), isOut(false), isSyncTimeout(false),
```

Comment:
I see your point but I don't agree that leaving them uninitialized is the solution. Initializing objects with safe defaults are the norm, if that makes the design harder to reason about it can be solved with comments as to what the fields does and what the values mean. 
A second option would be to not have parameter less constructors if there's no valid default values.